controllers/resource/reconciler.go

```diff
@@ -146,6 +151,37 @@ func (cor *clusterReconciler) Reconcile(ctx context.Context, objectKey types.Nam
 			return err
 		}
 		resources = append(resources, r...)
+	case anywherev1.NutanixDatacenterKind:
```

Comment:
Can you add a feature gate here similar to this? https://github.com/aws/eks-anywhere/blob/main/controllers/resource/reconciler.go#L109-L111